File: app/src/main/java/com/imperium/academio/ui/model/MaterialTopicRvModel.java
package com.imperium.academio.ui.model;

import androidx.annotation.Nullable;

public class MaterialTopicRvModel {
    private final String topic;

    public MaterialTopicRvModel(String topic) {
        this.topic = topic;
    }

    public String getTopic() {
        return topic;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (!(obj instanceof MaterialTopicRvModel))
            return false;
        return ((MaterialTopicRvModel) obj).topic.equals(topic);
    }
}

File: app/src/main/java/com/imperium/academio/ui/model/MaterialTopicRvModel.kt
package com.imperium.academio.ui.model

class MaterialTopicRvModel(val topic: String?) {
    override fun equals(other: Any?): Boolean {
        return if (other !is MaterialTopicRvModel) false else other.topic == topic
    }

    override fun hashCode(): Int {
        return topic.hashCode()
    }
}